devicetypes/smartthings/aeon-multiwhite-bulb.src/aeon-multiwhite-bulb.groovy

```diff
@@ -0,0 +1,223 @@
+}
+
+def installed() {
+	log.debug "installed()..."
+	sendEvent(name: "checkInterval", value: 1860, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "0"])
+	sendEvent(name: "level", value: 100, unit: "%")
+	sendEvent(name: "colorTemperature", value: 2700)
+}
+
+def parse(description) {
+	def result = null
+	if (description != "updated") {
+		def cmd = zwave.parse(description)
+		if (cmd) {
+			result = zwaveEvent(cmd)
+			log.debug("'$description' parsed to $result")
+		} else {
+			log.debug("Couldn't zwave.parse '$description'")
+		}
+	}
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	dimmerEvents(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicSet cmd) {
+	dimmerEvents(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelReport cmd) {
+	unschedule(offlinePing)
+	dimmerEvents(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchcolorv3.SwitchColorReport cmd) {
+	log.debug "got SwitchColorReport: $cmd"
+	def result = []
+	if (cmd.value == 255) {
+		def parameterNumber = (cmd.colorComponent == WARM_WHITE) ? WARM_WHITE_CONFIG : COLD_WHITE_CONFIG
+		result << response(command(zwave.configurationV2.configurationGet([parameterNumber: parameterNumber])))
+	}
+	result
+}
+
+private dimmerEvents(physicalgraph.zwave.Command cmd) {
+	def value = (cmd.value ? "on" : "off")
+	def result = [createEvent(name: "switch", value: value, descriptionText: "$device.displayName was turned $value")]
+	if (cmd.value) {
+		result << createEvent(name: "level", value: cmd.value == 99 ? 100 : cmd.value , unit: "%")
+	}
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	log.debug "got ConfigurationReport: $cmd"
+	def result = null
+	if (cmd.parameterNumber == WARM_WHITE_CONFIG || cmd.parameterNumber == COLD_WHITE_CONFIG)
+		result = createEvent(name: "colorTemperature", value: cmd.scaledConfigurationValue)
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	def linkText = device.label ?: device.name
+	[linkText: linkText, descriptionText: "$linkText: $cmd", displayed: false]
+}
+
+def buildOffOnEvent(cmd){
+	[zwave.basicV1.basicSet(value: cmd), zwave.switchMultilevelV3.switchMultilevelGet()]
+}
+
+def on() {
+	commands(buildOffOnEvent(0xFF), 5000)
+}
+
+def off() {
+	commands(buildOffOnEvent(0x00), 5000)
+}
+
+def refresh() {
+	commands([zwave.switchMultilevelV3.switchMultilevelGet()] + queryAllColors(), 500)
+}
+
+def ping() {
+	log.debug "ping().."
+	unschedule(offlinePing)
+	runEvery5Minutes(offlinePing)
```

Comment:
Is there a requirement behind pinging this device every 5 minutes while offline? Pinging an offline Z-Wave device can tie up the entire Z-Wave system for up to several seconds so we try not to do it too much.

Reply:
This is a good question and it was something that @greens and I discussed. This bulb doesn't Hail or notify in any way when it comes back online (e.g. power is restored). Since the device health ping has an increasing backoff (up to a few hours if left overnight), then it can take a really long time for this bulb to be considered online again. This offline ping is a bandaid to deal with that.

Reply:
I think the 5 minutes is @mckeed 's concern. I probably should have considered this too. Maybe 30m?

Reply:
Done.